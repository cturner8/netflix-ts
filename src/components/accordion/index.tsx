import { createContext, useContext, useState } from "react";
import {
  Inner,
  Container,
  Title,
  Frame,
  Item,
  Header,
  Body,
} from "./styles/accordion";

interface Props {}

interface TitleProps {}
interface FrameProps {}
interface ItemProps {}
interface HeaderProps {}
interface BodyProps {}

interface Compound {
  Title: React.FC<TitleProps>;
  Frame: React.FC<FrameProps>;
  Item: React.FC<ItemProps>;
  Header: React.FC<HeaderProps>;
  Body: React.FC<BodyProps>;
}

type ToggleShow = boolean;
type SetToggleShow = React.Dispatch<React.SetStateAction<ToggleShow>>;

interface Context {
  toggleShow?: ToggleShow;
  setToggleShow?: SetToggleShow;
}

const ToggleContext = createContext<Context>({});

const Accordion: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};

const ATitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const AFrame: React.FC<FrameProps> = ({ children, ...restProps }) => {
  return <Frame {...restProps}>{children}</Frame>;
};

const AItem: React.FC<ItemProps> = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

const AHeader: React.FC<HeaderProps> = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);
  const onClick = () => {
    if (setToggleShow) {
      setToggleShow((prevToggle: ToggleShow) => !prevToggle);
    }
  };
  const actionButton = toggleShow ? (
    <img src="/images/icons/close-slim.png" alt="close" />
  ) : (
    <img src="/images/icons/add.png" alt="open" />
  );

  return (
    <Header onClick={onClick} {...restProps}>
      {children}
      {actionButton}
    </Header>
  );
};

const ABody: React.FC<BodyProps> = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);

  return toggleShow ? <Body {...restProps}>{children}</Body> : <></>;
};

Accordion.Title = ATitle;
Accordion.Frame = AFrame;
Accordion.Item = AItem;
Accordion.Header = AHeader;
Accordion.Body = ABody;

export { Accordion };
