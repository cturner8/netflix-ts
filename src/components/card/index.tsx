import { createContext, useState, useContext, HTMLAttributes } from "react";
import {
  Container,
  Group,
  Title,
  SubTitle,
  Text,
  Meta,
  Item,
  Image,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
  Content,
  Entities,
} from "./styles/card";

interface Props {}

interface ItemFeature {}

export interface GroupProps {
  flexDirection: string;
  alignItems: string;
  margin: string;
}

interface TitleProps {}
interface SubTitleProps {}
interface TextProps {}
interface MetaProps {}
interface ItemProps extends HTMLAttributes<HTMLDivElement> {
  item: ItemFeature;
}
interface ImageProps {}
interface EntityProps {}
export interface FeatureProps {
  src: string;
}
export interface FeatureTextProps {
  fontWeight: string;
}
export interface MaturityProps {
  rating: number;
}

interface Context {
  showFeature: boolean;
  setShowFeature: Function;
  itemFeature: ItemFeature;
  setItemFeature: Function;
}

export const FeatureContext = createContext<Context>({
  showFeature: false,
  setShowFeature: () => {},
  itemFeature: {},
  setItemFeature: () => {},
});

interface Compound {
  Group: React.FC<GroupProps>;
  Title: React.FC<TitleProps>;
  SubTitle: React.FC<SubTitleProps>;
  Text: React.FC<TextProps>;
  Meta: React.FC<MetaProps>;
  Item: React.FC<ItemProps>;
  Image: React.FC<ImageProps>;
  Entities: React.FC<EntityProps>;
}

const Card: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...restProps}>{children}</Container>
    </FeatureContext.Provider>
  );
};

const CGroup: React.FC<GroupProps> = ({ children, ...restProps }) => {
  return <Group {...restProps}>{children}</Group>;
};

const CTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const CSubTitle: React.FC<SubTitleProps> = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

const CText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const CMeta: React.FC<MetaProps> = ({ children, ...restProps }) => {
  return <Meta {...restProps}>{children}</Meta>;
};

const CItem: React.FC<ItemProps> = ({ item, children, ...restProps }) => {
  const { setShowFeature, setItemFeature } = useContext(FeatureContext);

  return (
    <Item
      onClick={() => {
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...restProps}
    >
      {children}
    </Item>
  );
};

const CImage: React.FC<ImageProps> = ({ ...restProps }) => {
  return <Image {...restProps} />;
};

const CEntities: React.FC<EntityProps> = ({ children, ...restProps }) => {
  return <Entities {...restProps}>{children}</Entities>;
};

Card.Group = CGroup;
Card.Title = CTitle;
Card.SubTitle = CSubTitle;
Card.Text = CText;
Card.Meta = CMeta;
Card.Item = CItem;
Card.Image = CImage;
Card.Entities = CEntities;

export { Card };
