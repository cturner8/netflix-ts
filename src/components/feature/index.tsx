import { Container, Title, SubTitle } from "./styles/feature";

interface Props {}

interface TitleProps {}
interface SubTitleProps {}

interface Compound {
  Title: React.FC<TitleProps>;
  SubTitle: React.FC<SubTitleProps>;
}

const Feature: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FSubTitle: React.FC<SubTitleProps> = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Feature.Title = FTitle;
Feature.SubTitle = FSubTitle;

export { Feature };
