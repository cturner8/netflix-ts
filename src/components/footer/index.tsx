import {
  Container,
  Row,
  Column,
  Title,
  Link,
  Text,
  Break,
} from "./styles/footer";

export interface Props {}

interface RowProps {}
interface ColumnProps {}
interface TitleProps {}
interface LinkProps {
  href: string;
}
interface TextProps {}
interface BreakProps {}

interface Compound {
  Row: React.FC<RowProps>;
  Column: React.FC<ColumnProps>;
  Title: React.FC<TitleProps>;
  Link: React.FC<LinkProps>;
  Text: React.FC<TextProps>;
  Break: React.FC<BreakProps>;
}

const Footer: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FRow: React.FC<RowProps> = ({ children, ...restProps }) => {
  return <Row {...restProps}>{children}</Row>;
};

const FColumn: React.FC<ColumnProps> = ({ children, ...restProps }) => {
  return <Column {...restProps}>{children}</Column>;
};

const FLink: React.FC<LinkProps> = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

const FTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const FBreak: React.FC<BreakProps> = ({ children, ...restProps }) => {
  return <Break {...restProps}>{children}</Break>;
};

Footer.Row = FRow;
Footer.Column = FColumn;
Footer.Title = FTitle;
Footer.Link = FLink;
Footer.Text = FText;
Footer.Break = FBreak;

export { Footer };
