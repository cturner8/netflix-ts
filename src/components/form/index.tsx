import {
  InputHTMLAttributes,
  FormHTMLAttributes,
  ButtonHTMLAttributes,
} from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

interface Props {}

interface ErrorProps {}
interface BaseProps extends FormHTMLAttributes<HTMLFormElement> {}
interface TitleProps {}
interface TextProps {}
interface TextSmallProps {}
interface LinkProps {
  to: string;
}
interface InputProps extends InputHTMLAttributes<HTMLInputElement> {}
interface SubmitProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

interface Compound {
  Error: React.FC<ErrorProps>;
  Base: React.FC<BaseProps>;
  Title: React.FC<TitleProps>;
  Text: React.FC<TextProps>;
  TextSmall: React.FC<TextSmallProps>;
  Link: React.FC<LinkProps>;
  Input: React.FC<InputProps>;
  Submit: React.FC<SubmitProps>;
}

const Form: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FError: React.FC<ErrorProps> = ({ children, ...restProps }) => {
  return <Error {...restProps}>{children}</Error>;
};

const FBase: React.FC<BaseProps> = ({ children, ...restProps }) => {
  return <Base {...restProps}>{children}</Base>;
};

const FTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const FTextSmall: React.FC<TextSmallProps> = ({ children, ...restProps }) => {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

const FLink: React.FC<LinkProps> = ({ children, to, ...restProps }) => {
  return (
    <Link to={to} {...restProps}>
      {children}
    </Link>
  );
};

const FInput: React.FC<InputProps> = ({ children, ...restProps }) => {
  return <Input {...restProps}>{children}</Input>;
};

const FSubmit: React.FC<SubmitProps> = ({ children, ...restProps }) => {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Error = FError;
Form.Base = FBase;
Form.Title = FTitle;
Form.Text = FText;
Form.TextSmall = FTextSmall;
Form.Link = FLink;
Form.Input = FInput;
Form.Submit = FSubmit;

export { Form };
