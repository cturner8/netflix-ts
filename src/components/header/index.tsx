import {
  ImgHTMLAttributes,
  HTMLAttributes,
  useState,
  Dispatch,
  SetStateAction,
  InputHTMLAttributes,
} from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  Search,
  SearchInput,
  SearchIcon,
  PlayButton,
} from "./styles/header";

export interface Props {
  bg?: boolean;
  src?: string;
  dontShowSmallViewPort?: boolean;
}

interface FrameProps {}
interface GroupProps {}
export interface PictureProps {
  src: string | null | undefined;
}
interface ProfileProps {}
interface LogoProps extends ImgHTMLAttributes<HTMLImageElement> {
  to: string;
}
interface ButtonLinkProps {
  to: string;
}

interface DropdownProps {}
interface FeatureProps {}
interface TextProps {}
export interface TextLinkProps extends HTMLAttributes<HTMLParagraphElement> {
  active: "true" | "false";
}

interface FeatureCallOutProps {}

type SearchTerm = string;
type SetSearchTerm = Dispatch<SetStateAction<SearchTerm>>;
interface SearchTermProps {
  searchTerm: SearchTerm;
  setSearchTerm: SetSearchTerm;
}

export interface SearchInputProps
  extends InputHTMLAttributes<HTMLInputElement> {
  active: boolean;
}

interface PlayButtonProps {}

interface Compound {
  Frame: React.FC<FrameProps>;
  Group: React.FC<GroupProps>;
  Logo: React.FC<LogoProps>;
  ButtonLink: React.FC<ButtonLinkProps>;
  Feature: React.FC<FeatureProps>;
  Text: React.FC<TextProps>;
  TextLink: React.FC<TextLinkProps>;
  FeatureCallOut: React.FC<FeatureCallOutProps>;
  Picture: React.FC<PictureProps>;
  Profile: React.FC<ProfileProps>;
  Dropdown: React.FC<DropdownProps>;
  SearchTerm: React.FC<SearchTermProps>;
  PlayButton: React.FC<PlayButtonProps>;
}

const Header: React.FC<Props> & Compound = ({
  bg = true,
  children,
  ...restProps
}) => {
  return bg ? (
    <Background {...restProps}>{children}</Background>
  ) : (
    <>{children}</>
  );
};

const HFrame: React.FC<FrameProps> = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const HGroup: React.FC<GroupProps> = ({ children, ...restProps }) => {
  return <Group {...restProps}>{children}</Group>;
};

const HPicture: React.FC<PictureProps> = ({ src, ...restProps }) => {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

const HProfile: React.FC<ProfileProps> = ({ children, ...restProps }) => {
  return <Profile {...restProps}>{children}</Profile>;
};

const HSearchTerm: React.FC<SearchTermProps> = ({
  searchTerm,
  setSearchTerm,
  ...restProps
}) => {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive((prevActive) => !prevActive)}>
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
      />
    </Search>
  );
};

const HLogo: React.FC<LogoProps> = ({ to, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

const HButtonLink: React.FC<ButtonLinkProps> = ({ ...restProps }) => {
  return <ButtonLink {...restProps} />;
};

const HFeatureCallOut: React.FC<FeatureCallOutProps> = ({
  children,
  ...restProps
}) => {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

const HFeature: React.FC<FeatureProps> = ({ children, ...restProps }) => {
  return <Feature {...restProps}>{children}</Feature>;
};

const HDropdown: React.FC<DropdownProps> = ({ children, ...restProps }) => {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

const HText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const HTextLink: React.FC<TextLinkProps> = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

const HPlayButton: React.FC<PlayButtonProps> = ({ children, ...restProps }) => {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Frame = HFrame;
Header.Group = HGroup;
Header.Logo = HLogo;
Header.ButtonLink = HButtonLink;
Header.Feature = HFeature;
Header.Text = HText;
Header.TextLink = HTextLink;
Header.FeatureCallOut = HFeatureCallOut;
Header.Picture = HPicture;
Header.Profile = HProfile;
Header.Dropdown = HDropdown;
Header.SearchTerm = HSearchTerm;
Header.PlayButton = HPlayButton;

export { Header };
