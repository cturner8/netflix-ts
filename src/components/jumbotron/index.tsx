import {
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
  Item,
} from "./styles/jumbotron";

export interface Props {
  direction?: string;
}

interface ContainerProps {}
interface PaneProps {}
interface TitleProps {}
interface SubTitleProps {}
interface ImageProps {
  src: string;
  alt: string;
}

interface Compound {
  Container: React.FC<ContainerProps>;
  Pane: React.FC<PaneProps>;
  Title: React.FC<TitleProps>;
  SubTitle: React.FC<SubTitleProps>;
  Image: React.FC<ImageProps>;
}

const Jumbotron: React.FC<Props> & Compound = ({
  children,
  direction = "row",
  ...restProps
}) => {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
};

const JContainer: React.FC<ContainerProps> = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const JPane: React.FC<PaneProps> = ({ children, ...restProps }) => {
  return <Pane {...restProps}>{children}</Pane>;
};

const JTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const JSubTitle: React.FC<SubTitleProps> = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

const JImage: React.FC<ImageProps> = ({ ...restProps }) => {
  return <Image {...restProps} />;
};

Jumbotron.Container = JContainer;
Jumbotron.Pane = JPane;
Jumbotron.Title = JTitle;
Jumbotron.SubTitle = JSubTitle;
Jumbotron.Image = JImage;

export { Jumbotron };
