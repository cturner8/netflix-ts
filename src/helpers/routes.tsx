import { Route, Redirect, RouteProps } from "react-router-dom";
import * as routes from "../constants/routes";
import { FirebaseUser } from "../lib/firebase";

interface IsUserRedirectProps extends RouteProps {
  user: FirebaseUser;
  loggedInPath: string;
}

interface ProtectedRouteProps extends RouteProps {
  user: FirebaseUser;
}

export const IsUserRedirect: React.FC<IsUserRedirectProps> = ({
  user,
  loggedInPath,
  children,
  ...rest
}) => {
  return (
    <Route
      {...rest}
      render={() =>
        !user ? <>{children}</> : <Redirect to={{ pathname: loggedInPath }} />
      }
    />
  );
};

export const ProtectedRoute: React.FC<ProtectedRouteProps> = ({
  user,
  children,
  ...rest
}) => {
  return (
    <Route
      {...rest}
      render={({ location }) =>
        user ? (
          <>{children}</>
        ) : (
          <Redirect
            to={{ pathname: routes.SIGN_IN, state: { from: location } }}
          />
        )
      }
    />
  );
};
