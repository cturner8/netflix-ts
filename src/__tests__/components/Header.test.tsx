import { render, fireEvent } from "@testing-library/react";
import { Header } from "../../components";
import * as routes from "../../constants/routes";

jest.mock("react-router-dom", () => ({
  Link: () => <></>,
}));

describe("<Header />", () => {
  it("renders the basic header with a background", () => {
    const { container, getByText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" to="/" />
          <Header.TextLink active="true">Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText(/Hello I am a link!/i)).toBeTruthy();
    expect(getByTestId("header-bg")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the basic header without a background", () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" to="/" />
          <Header.ButtonLink to="/">Sign In</Header.ButtonLink>
          <Header.TextLink active="false">Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText(/Hello I am a link!/i)).toBeTruthy();
    expect(queryByTestId("header-bg")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the full header with a background", () => {
    const { container, getByText, getByTestId } = render(
      <Header dontShowSmallViewPort src={"joker1"}>
        <Header.Frame>
          <Header.Group>
            <Header.Logo
              to={routes.HOME}
              src={"/images/logo.svg"}
              alt="Netflix"
            />
            <Header.TextLink active={"false"}>Series</Header.TextLink>
            <Header.TextLink active={"true"}>Films</Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.SearchTerm searchTerm={"joker"} setSearchTerm={() => {}} />
            <Header.Profile>
              <Header.Picture src={"/images/cameron.png"} />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src={"/images/cameron.png"} />
                  <Header.TextLink active="false">Cameron</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink active="false">Sign out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
          <Header.Text>Forever alone in a crowd...</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>
    );

    const searchInput = getByTestId("search-input") as HTMLInputElement;

    expect(searchInput).toBeTruthy();
    expect(searchInput.value).toBe("joker");
    fireEvent.change(searchInput, { target: { value: "Simpsons" } });
    fireEvent.click(getByTestId("search-click"));

    expect(getByText(/Series/i)).toBeTruthy();
    expect(getByText(/Films/i)).toBeTruthy();
    expect(getByText(/Cameron/i)).toBeTruthy();
    expect(getByText(/Watch Joker Now/i)).toBeTruthy();
    expect(getByText(/Sign Out/i)).toBeTruthy();
    expect(getByText(/Play/i)).toBeTruthy();
    expect(getByText(/Forever alone in a crowd.../i)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
