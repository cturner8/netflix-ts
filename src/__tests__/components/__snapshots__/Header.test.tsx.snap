// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic header with a background 1`] = `
<div
  class="header__Background-o8frjl-0 eJSPXA"
  data-testid="header-bg"
>
  <div
    class="header__Container-o8frjl-1 bwqZsE"
  >
    <p
      class="header__Link-o8frjl-7 ebSJlD"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic header without a background 1`] = `
<div
  class="header__Container-o8frjl-1 bwqZsE"
>
  <p
    class="header__Link-o8frjl-7 hPHHGD"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header /> renders the full header with a background 1`] = `
<div
  class="header__Background-o8frjl-0 fagfLB"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-o8frjl-1 bwqZsE"
  >
    <div
      class="header__Group-o8frjl-8 cZWEOE"
    >
      <p
        class="header__Link-o8frjl-7 hPHHGD"
      >
        Series
      </p>
      <p
        class="header__Link-o8frjl-7 ebSJlD"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-o8frjl-8 cZWEOE"
    >
      <div
        class="header__Search-o8frjl-12 jJevlh"
      >
        <button
          class="header__SearchIcon-o8frjl-13 iZnaJy"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-o8frjl-14 gjOCLx"
          data-testid="search-input"
          placeholder="Search films and series"
          value="joker"
        />
      </div>
      <div
        class="header__Profile-o8frjl-11 cQoCKg"
      >
        <button
          class="header__Picture-o8frjl-9 kwudvj"
          src="/images/users//images/cameron.png.png"
        />
        <div
          class="header__Dropdown-o8frjl-10 fDnCnf"
        >
          <div
            class="header__Group-o8frjl-8 cZWEOE"
          >
            <button
              class="header__Picture-o8frjl-9 kwudvj"
              src="/images/users//images/cameron.png.png"
            />
            <p
              class="header__Link-o8frjl-7 hPHHGD"
            >
              Cameron
            </p>
          </div>
          <div
            class="header__Group-o8frjl-8 cZWEOE"
          >
            <p
              class="header__Link-o8frjl-7 hPHHGD"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-o8frjl-1 header__Feature-o8frjl-4 bwqZsE FBBwe"
  >
    <h2
      class="header__FeatureCallOut-o8frjl-6 kDa-DuO"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-o8frjl-5 kVXHcX"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-o8frjl-15 lekdEZ"
    >
      Play
    </button>
  </div>
</div>
`;
